### PR TITLE
Gutenboarding: move design picker state in onboarding store

### PR DESCRIPTION
**Pull request by razvanpapadopol**
#### Changes proposed in this Pull Request

* add selected page layouts to onboarding store
* <s>[WIP] add selected design to store</s> Already [implemented]( https://github.com/Automattic/wp-calypso/pull/38705/files#diff-b43d337b3bf37258ebc7f894a3e9b6a5L48-R48) in https://github.com/Automattic/wp-calypso/pull/38705.

#### Testing instructions
* Selected page layouts should be persisted during one visit (including when selecting a different design or navigating to previous step and back)

Fixes one issue mentioned in https://github.com/Automattic/wp-calypso/pull/38583#pullrequestreview-337251522
> Esc cancels all page selections and returns to design picker, just like clicking outside page selector does as well. I expected my page selections to remain tho; they got wiped out.

### CODE REVIEW

**Comment by chriskmnds** [comment]
LGTM 

1. selected pages
2. scrolled/open different designs - same pages selected 
3. went back/forth - same pages selected

File: client/landing/gutenboarding/stores/onboard/reducer.ts [hunk 1]
```diff
@@ -43,7 +43,24 @@ const siteVertical: Reducer<
 	return state;
 };
 
-const reducer = combineReducers( { domain, siteTitle, siteVertical } );
+const pageLayouts: Reducer< Set< string >, ReturnType< typeof Actions[ 'togglePageLayout' ] > > = (
+	state = new Set(),
```
Comment (sirreal) [inline client/landing/gutenboarding/stores/onboard/reducer.ts]: I suspect we'll end up persisting this, `Set` won't serialize well, and we don't have fine-grained control over the wp-data persistence behavior.

Shall we add `pageLayouts` to the persistence array and see how this behaves?

Comment (razvanpapadopol) [reply client/landing/gutenboarding/stores/onboard/reducer.ts]: Good catch. It should work now with `Array`. I've set it to persist and we can remove later if needed.

Comment (chriskmnds) [reply client/landing/gutenboarding/stores/onboard/reducer.ts]: + good to know :-)

File: client/landing/gutenboarding/stores/onboard/reducer.ts [hunk 2]
```diff
@@ -43,7 +43,21 @@ const siteVertical: Reducer<
...
-const reducer = combineReducers( { domain, siteTitle, siteVertical } );
+const pageLayouts: Reducer<
+	Array< string >,
+	ReturnType< typeof Actions[ 'togglePageLayout' ] >
+> = ( state = [], action ) => {
+	if ( action.type === ActionType.TOGGLE_PAGE_LAYOUT ) {
+		const layout = action.pageLayout;
+		if ( state.includes( layout.slug ) ) {
+			return state.filter( item => item !== layout.slug );
+		}
```
Comment (chriskmnds) [inline client/landing/gutenboarding/stores/onboard/reducer.ts]: p.s. (probably worth ignoring) but would it help to reduce this down to `O(N)`, or a single pass of the array? Something like:

```
const filtered = state.filter( item => item !== layout.slug );

if ( state.length === filtered.length ) {
    return [ ...state, item ];
}

return filtered;
```

or is that just harder to read with little extra value? Maybe there's a cleaner approach?

File: client/landing/gutenboarding/stores/onboard/actions.ts [hunk 3]
```diff
@@ -23,3 +30,8 @@ export const setSiteTitle = ( siteTitle: string ) => ( {
 	type: ActionType.SET_SITE_TITLE as const,
 	siteTitle,
 } );
+
+export const togglePageLayout = ( pageLayout: Template ) => ( {
```
Comment (sirreal) [inline client/landing/gutenboarding/stores/onboard/actions.ts]: I don't like `toggle` as an action and prefer `add`/`remove`, but this is fine for now 🙂

File: client/landing/gutenboarding/stores/onboard/reducer.ts (hunk 2 above)
Comment (sirreal) [reply client/landing/gutenboarding/stores/onboard/reducer.ts]: This probably isn't worth optimizing much.

Comment (chriskmnds) [reply client/landing/gutenboarding/stores/onboard/reducer.ts]: Heh I guess suggesting to improve the state tree into an indexed form may sound even less indulging.

**Pull request merged**